src/api/block.rs

```diff
@@ -67,460 +66,477 @@ impl BlockApi {
         &self,
         req: BlockTransactionRequest,
     ) -> ApiResult<BlockTransactionResponse> {
-        let block_hash = block_hash_from_string(req.block_identifier.hash.as_str())?;
-        let block_summary = self
-            .query_helper
-            .query_block_summary_by_hash(&block_hash) // TODO should probably use the "raw" variant
-            .await ?;
-        match block_summary
-            .transaction_summaries()
-            .iter()
-            .find(|t| t.hash.to_string() == req.transaction_identifier.hash)
-        {
-            None => Err(ApiError::NoTransactionsMatched),
-            Some(transaction) => Ok(BlockTransactionResponse::new(map_transaction(transaction))),
+        let tx_status =
+            self.query_helper.query_transaction_status(req.transaction_identifier.hash).await?;
```

Comment:
The request also has a block that is not being used now. Ideally we ought to check that it's in the specified block. Is that doable? Otherwise we should mention in the docs that we don't use this data.

This is not important to fix in this PR; a TODO and/or task is sufficient.